### PR TITLE
70 allow buildings to be built

### LINKED ISSUE #70: Allow buildings to be built
In the system view, it should show buildings that are available to be built. 
Create a sudo-backend for this data. Contact me if you want some basic building ideas and a few categories

### PR DESCRIPTION
**Pull request by PaulMasonDev**
@DylanBozarth @linuxuser07 Please DO NOT merge into main yet. There are some refinements I want to make to the branch before merging (and there is a jest issue, it broke somewhere after implementation and I need to fix it).

A couple things I may need from you guys:

1. Ideas for the building types and what the build requirements are. I'll make some annotations in the commits as to what the code does.  I'm hoping its self-explanatory, but just in case.
2. Scenarios that I can write jest test cases for based upon the building types we come up with. I think it will be important to write tests for this system to ensure we don't break something along the way.

In short, I think I did a singleton pattern with a class (Fernando can check me on that). the BuildingManager class will just have a single instance created which is exported and could be used throughout the app. I haven't made any implementations at the component level where it might be used, so I just called the getter method from the class on App.tsx for now.

Future work will include tightening/DRYing up the logic I have created, and then seeing if I can get Jest back on track.

### CODE REVIEW

File: src/pseudo-backend/buildings/Buildings.constants.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { BuildingsMapping, PlanetTypes } from './Buildings.types';
+
+export const buildingsMapping: BuildingsMapping = {
```
Comment (PaulMasonDev) [inline src/pseudo-backend/buildings/Buildings.constants.ts:3]: @DylanBozarth @linuxuser07 This is where we will define the requirements for each building. I've made the typings pretty strict here on initial development, but we can loosen them up later if we want to.  We can add whatever we want here including min/max resources from the system, and system types that we want to include or exclude. (Right now its something like "Radioactive" and "Gaia" but we could add other things like "Under Revolution", "Under Coup" for example if we want to do some checks that way.

File: src/App.tsx [hunk 2]
```diff
@@ -7,8 +7,16 @@ import { SectorA } from './views/sectors/sector-a';
 import { DetailsMenu } from './UI/menu-pages/detailsMenu';
 import Login from './components/Authenication/login';
 import Register from './components/Authenication/register';
+import buildingManager from './pseudo-backend';
+import { PlanetTypes } from './pseudo-backend/buildings/Buildings.types';
 
 function App() {
+  buildingManager.getAvailableBuildings({
```
Comment (PaulMasonDev) [inline src/App.tsx]: I envision us actually getting this information from the backend at some point, but for now, this should show what I have in mind.

File: src/pseudo-backend/buildings/BuildingManager.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { BuildingChecks, Buildings, BuildingsMapping, SystemValues } from './Buildings.types';
```
Comment (PaulMasonDev) [inline src/pseudo-backend/buildings/BuildingManager.ts:1]: Our first tech debt ticket!  https://github.com/DylanBozarth/Andromeda/issues/74

**Comment by PaulMasonDev** [comment]
@DylanBozarth @linuxuser07 Should now be ready to merge. Next steps would be to create issues around implementation based on actual values, and then fleshing out the building types so that information can be added to the constant file.

**Comment by PaulMasonDev** [comment]
@DylanBozarth So I realized I didn't need to do anything else until we get some defined requirements of buildings and how they relate to the properties provided by a player system (See my comments from when I opened the PR).  So there is no implementation in the game yet, but the psuedo backend here should support whatever we decide. Let's just leave the test file in so we have an example to go on. But we need to come up with building definitions (let me know if they are already defined somewhere).

**Comment by PaulMasonDev** [comment]
Opening new PR due to staleness of code.
